packages/utils/src/useTableTools/FilterConfigBuilder/FilterBuilder.js

```diff
@@ -0,0 +1,47 @@
+import { conditionalFilterType } from '@redhat-cloud-services/frontend-components/ConditionalFilter';
+import { stringToId } from '../helper';
+
+class FilterBuilder {
+  constructor(filterConfig) {
+    this.filterConfig = filterConfig;
+    this.config = this.filterConfig.config;
+  }
+
+  buildFilterFilterString = (configItem, value) => {
+    const { type, filterString } = configItem;
+
+    if (type !== 'hidden' && !value) {
+      return [];
+    }
+
+    switch (type) {
+      case conditionalFilterType.text:
+        return [filterString(value)];
+      case conditionalFilterType.checkbox:
+        return value.map((filter) => filterString(filter));
+
+      case conditionalFilterType.group:
+        return filterString(value);
+
+      case 'hidden':
+        return filterString();
+
+      default:
+        return [];
+    }
+  };
+
+  combineFilterStrings = (filterStringArray) => {
+    const moreThanTwo = filterStringArray.map((f) => f.length).filter((fl) => fl > 0).length >= 2;
+    return filterStringArray.map((fs) => fs.join(' or ')).join(moreThanTwo ? ' and ' : '');
+  };
+
+  buildFilterString = (filters) => {
```

Comment:
This filter builder can probably go back into Compliance as helper functions. I'll look into that. 

The idea is/was that filters have a function/config to prepare the filter values to query an API, but this can be implemented to be more generic, I believe.

Reply:
This is now removed from here and will be implemented in Compliance itself.